samba4-lib: compile error

Description:

On Raspberry 1~4  four devices, all of them have the same problom:

I add  `CONFIG_PACKAGE_libunwind=y` but the error still presence:

```
2020-06-14T06:02:58.6546049Z make[2]: Entering directory '/home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/feeds/packages/net/samba4'
2020-06-14T06:02:58.9485131Z mkdir -p /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/bin/targets/bcm27xx/bcm2711/packages /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/build_dir/target-aarch64_cortex-a72_musl/samba-4.12.3/ipkg-aarch64_cortex-a72/samba4-libs/CONTROL /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/staging_dir/target-aarch64_cortex-a72_musl/pkginfo
2020-06-14T06:02:58.9511141Z install -d -m0755 /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/build_dir/target-aarch64_cortex-a72_musl/samba-4.12.3/ipkg-aarch64_cortex-a72/samba4-libs/usr/lib
2020-06-14T06:02:58.9535878Z cp -fpR /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/build_dir/target-aarch64_cortex-a72_musl/samba-4.12.3/ipkg-install/usr/lib/*.so* /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/build_dir/target-aarch64_cortex-a72_musl/samba-4.12.3/ipkg-aarch64_cortex-a72/samba4-libs/usr/lib/
2020-06-14T06:02:58.9652266Z # rpath-install
2020-06-14T06:02:58.9669605Z cp -fpR /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/build_dir/target-aarch64_cortex-a72_musl/samba-4.12.3/ipkg-install/usr/lib/samba /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/build_dir/target-aarch64_cortex-a72_musl/samba-4.12.3/ipkg-aarch64_cortex-a72/samba4-libs/usr/lib/
2020-06-14T06:02:58.9902464Z find /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/build_dir/target-aarch64_cortex-a72_musl/samba-4.12.3/ipkg-aarch64_cortex-a72/samba4-libs -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| xargs -r rm -rf
2020-06-14T06:02:59.3412916Z Package samba4-libs is missing dependencies for the following libraries:
2020-06-14T06:02:59.3419753Z libunwind-aarch64.so.8
2020-06-14T06:02:59.3425719Z Makefile:449: recipe for target '/home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/bin/packages/aarch64_cortex-a72/packages/samba4-libs_4.12.3-1_aarch64_cortex-a72.ipk' failed
2020-06-14T06:02:59.3426433Z make[2]: Leaving directory '/home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/feeds/packages/net/samba4'
2020-06-14T06:02:59.3434644Z make[2]: *** [/home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/bin/packages/aarch64_cortex-a72/packages/samba4-libs_4.12.3-1_aarch64_cortex-a72.ipk] Error 1
2020-06-14T06:02:59.3434839Z time: package/feeds/packages/samba4/compile#0.57#0.28#0.69
2020-06-14T06:02:59.3470882Z make[1]: *** [package/feeds/packages/samba4/compile] Error 2
2020-06-14T06:02:59.3471219Z package/Makefile:111: recipe for target 'package/feeds/packages/samba4/compile' failed
2020-06-14T06:02:59.3471612Z make[1]: Leaving directory '/home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt'
2020-06-14T06:02:59.3481574Z /home/runner/work/OpenWrt-Rpi/OpenWrt-Rpi/openwrt/include/toplevel.mk:222: recipe for target 'package/compile' failed
2020-06-14T06:02:59.3488043Z make: *** [package/compile] Error 2
2020-06-14T06:02:59.3530141Z ##[error]Process completed with exit code 2.
```

There are full logs:

<https://github.com/SuLingGG/OpenWrt-Rpi/runs/769196082?check_suite_focus=true>

--- a/net/samba4/Makefile
+++ b/net/samba4/Makefile
@@ -3,7 +3,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=samba
 PKG_VERSION:=4.12.3
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:= \

--- a/net/samba4/patches/103-samba-4.12-unbundle-libunwind.patch
+++ b/net/samba4/patches/103-samba-4.12-unbundle-libunwind.patch
@@ -1,0 +1,23 @@
+--- a/lib/util/wscript_configure	2020-06-14
++++ b/lib/util/wscript_configure	2020-06-14
+@@ -11,13 +11,13 @@ conf.CHECK_FUNCS_IN('backtrace backtrace
+ conf.CHECK_HEADERS('execinfo.h')
+ 
+ conf.SET_TARGET_TYPE('LIBUNWIND', 'EMPTY')
+-if conf.check_cfg(package='libunwind-generic',
+-                  args='--cflags --libs',
+-                  msg='Checking for libunwind',
+-                  uselib_store='LIBUNWIND',
+-                  mandatory=False):
+-    if conf.CHECK_HEADERS('libunwind.h'):
+-        conf.SET_TARGET_TYPE('LIBUNWIND', 'SYSLIB')
++# if conf.check_cfg(package='libunwind-generic',
++                  # args='--cflags --libs',
++                  # msg='Checking for libunwind',
++                  # uselib_store='LIBUNWIND',
++                  # mandatory=False):
++    # if conf.CHECK_HEADERS('libunwind.h'):
++        # conf.SET_TARGET_TYPE('LIBUNWIND', 'SYSLIB')
+ 
+ conf.CHECK_STRUCTURE_MEMBER('struct statvfs', 'f_frsize', define='HAVE_FRSIZE', headers='sys/statvfs.h')
+ 

--- a/net/samba4/patches/104-samba-4.12-unbundle-icu.patch
+++ b/net/samba4/patches/104-samba-4.12-unbundle-icu.patch
@@ -1,0 +1,30 @@
+--- a/lib/util/charset/wscript_configure	2020-06-14
++++ b/lib/util/charset/wscript_configure	2020-06-14
+@@ -37,15 +37,15 @@ conf.CHECK_CODE('''
+                 lib='iconv',
+                 headers='errno.h iconv.h')
+ 
+-if conf.CHECK_CFG(package='icu-i18n',
+-               args='--cflags --libs',
+-               msg='Checking for icu-i18n',
+-               uselib_store='ICU_I18N'):
+-    for lib in conf.env['LIB_ICU_I18N']:
+-        conf.CHECK_LIB(lib, shlib=True, mandatory=True)
+-    conf.env['icu-libs'] = ' '.join(conf.env['LIB_ICU_I18N'])
+-    if not conf.CHECK_HEADERS('unicode/ustring.h'):
+-        conf.fatal('Found libicu, but unicode/ustring.h is missing')
+-    conf.DEFINE('HAVE_UTF8_NORMALISATION', 1)
+-else:
+-    conf.env['icu-libs'] = ''
++# if conf.CHECK_CFG(package='icu-i18n',
++               # args='--cflags --libs',
++               # msg='Checking for icu-i18n',
++               # uselib_store='ICU_I18N'):
++    # for lib in conf.env['LIB_ICU_I18N']:
++        # conf.CHECK_LIB(lib, shlib=True, mandatory=True)
++    # conf.env['icu-libs'] = ' '.join(conf.env['LIB_ICU_I18N'])
++    # if not conf.CHECK_HEADERS('unicode/ustring.h'):
++        # conf.fatal('Found libicu, but unicode/ustring.h is missing')
++    # conf.DEFINE('HAVE_UTF8_NORMALISATION', 1)
++# else:
++conf.env['icu-libs'] = ''

